CrazyCanvas/Source/World/LevelObjectCreator.cpp

```diff
@@ -195,11 +200,74 @@ ESpecialObjectType LevelObjectCreator::CreateSpawnpoint(const LambdaEngine::Spec
 
 ESpecialObjectType LevelObjectCreator::CreateFlag(const LambdaEngine::SpecialObjectOnLoad& specialObject, LambdaEngine::TArray<LambdaEngine::Entity>& createdEntities, const glm::vec3& translation)
 {
-	UNREFERENCED_VARIABLE(specialObject);
-	UNREFERENCED_VARIABLE(createdEntities);
-	UNREFERENCED_VARIABLE(translation);
+	using namespace LambdaEngine;
+
+	if (specialObject.MeshComponents.GetSize() > 1)
+	{
+		LOG_WARNING("[LevelObjectCreator]: Special Object \"Flag\" has more than one Mesh Component, this is not currently supported");
+	}
+
+	const MeshComponent& meshComponent = specialObject.MeshComponents[0];
+
+	ECSCore* pECS					= ECSCore::GetInstance();
+	PhysicsSystem* pPhysicsSystem	= PhysicsSystem::GetInstance();
+
+	Entity entity = pECS->CreateEntity();
+
+	pECS->AddComponent<OffsetComponent>(entity, OffsetComponent{ .Offset = glm::vec3(0.0f)});
+	pECS->AddComponent<ParentComponent>(entity, ParentComponent{ .Attached = false });
+	const CollisionCreateInfo collisionCreateInfo =
+	{
+		.Entity				= entity,
+		.Position			= pECS->AddComponent<PositionComponent>(entity,		{ true, specialObject.DefaultPosition/* + translation + glm::vec3(0.0f, specialObject.DefaultScale.y * specialObject.BoundingBoxes[0].HalfExtent.y, 0.0f)*/}),
```

Comment:
Did you forget to remove a temporary comment?

Reply:
Yes, thanks!